napari/_vispy/canvas.py

```diff
@@ -1,36 +1,96 @@
 """VispyCanvas class.
 """
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
 from weakref import WeakSet
 
-from vispy.scene import SceneCanvas, Widget
+import numpy as np
+from vispy.scene import SceneCanvas as SceneCanvas_
+from vispy.scene import Widget
 
+from napari._vispy import VispyCamera
 from napari._vispy.utils.gl import get_max_texture_sizes
+from napari._vispy.utils.visual import create_vispy_overlay
+from napari.components.overlays import CanvasOverlay, SceneOverlay
+from napari.utils._proxies import ReadOnlyWrapper
 from napari.utils.colormaps.standardize_color import transform_color
+from napari.utils.interactions import (
+    mouse_double_click_callbacks,
+    mouse_move_callbacks,
+    mouse_press_callbacks,
+    mouse_release_callbacks,
+    mouse_wheel_callbacks,
+)
+
+if TYPE_CHECKING:
+    from qtpy.QtGui import QCursor, QImage
+
+    from napari.components import ViewerModel
+    from napari.components.overlays import Overlay
+
+
+class SceneCanvas(SceneCanvas_):
```

Comment:
What is the benefit of using the name `SceneCanvas` instead of something like `NapariSceneCanvas` that will simplify searching codebase?

Reply:
I wouldn't say there is a particular benefit. Right now the only usage of SceneCanvas is in _vispy/canvas.py though. I can change it but would it really simplify the searching codebase?

Reply:
Yes. As many tools for searching references by name may return the wrong result in the first place. For example, returning napari version instead of the original one.

Reply:
Resolved with [f8f02c7](https://github.com/napari/napari/pull/5432/commits/f8f02c7fe4e2b83a34d951381b95a68283101058)